var/spack/repos/builtin/packages/mercury/package.py

```diff
@@ -9,21 +9,21 @@ class Mercury(CMakePackage):
     """Mercury is a C library for implementing RPC, optimized for HPC"""
 
     homepage = 'https://mercury-hpc.github.io/'
-    url = 'https://github.com/mercury-hpc/mercury/archive/0.9.0.tar.gz'
+    url = 'https://github.com/mercury-hpc/mercury/releases/download/v1.0.1/mercury-1.0.1.tar.bz2'
     git = 'https://github.com/mercury-hpc/mercury.git'
 
     version('develop', branch='master', submodules=True)
-    version('1.0.1', tag='v1.0.1', submodules=True)
-    version('1.0.0', tag='v1.0.0', submodules=True)
-    version('0.9.0', tag='v0.9.0', submodules=True)
-
-    variant('bmi', default=False, description='Use BMI for network transport')
-    variant('cci', default=False, description='Use CCI for network transport')
-    variant('mpi', default=False, description='Use MPI for network transport')
-    variant('ofi', default=True,  description='Use libfabric plugin')
+    version('1.0.1', preferred=True, sha256='02febd56c401ef7afa250caf28d012b37dee842bfde7ee16fcd2f741b9cf25b3')
+    version('1.0.0', sha256='fb0e44d13f4652f53e21040435f91d452bc2b629b6e98dcf5292cd0bece899d4')
+    version('0.9.0', sha256='40868e141cac035213fe79400f8926823fb1f5a0651fd7027cbe162b063843ef')
+
+    variant('bmi', default=False, description='Use BMI plugin')
+    variant('cci', default=False, description='Use CCI plugin')
+    variant('mpi', default=False, description='Use MPI plugin')
+    variant('ofi', default=True,  description='Use OFI libfabric plugin')
     variant('sm',  default=False, description='Use shared-memory plugin')
-    variant('opa', default=False, description='Use OpenPA for atomics')
-    variant('boost', default=True, description='Use BOOST preprocessor macros')
+    variant('boostsys', default=False, description='Use system-installed Boost libraries')
```

Comment:
We generally avoid linking to system dependencies, as they result in unreproducible builds. Do you think it would be okay to remove this variant and always depend on boost?

Reply:
I understand but it is not using the "system" boost in that case. Mercury has two options: either it provides and installs the boost preprocessor headers that are included with mercury (so that there is no need to install boost) or in the case where boost is already installed, it won't install the boost prepocessor headers and just use the ones from the boost dependency. So when using the `+boostsys` variant, it will use/install the boost dependency and when not using that variant there won't be any dependency on boost and the headers included with mercury will be installed along with the mercury headers.

Reply:
Ah, I see. In that case, this is probably fine, although you may want to rename the description to make it clear that you are using a Spack-installed copy and not a copy installed on the system. Still, there might be some benefit to always depending on boost as you will always have an up-to-date copy of the headers, although I understand that it will increase installation times. I don't personally use `mercury` (I was simply updating the package for @mlawsonca) so I'll defer to your opinion on this one.

Reply:
OK :) I could simply default that variant to true then so that we always depend on boost by default and that'll give a chance to someone who does not want to install boost to use the headers that we provide. I will also update the description to make sure it's clear. Thanks.